File: Array/ways-to-make-a-fair-array.cpp
class Solution {
public:
    int waysToMakeFair(vector<int>& nums) {
         int even=0,odd=0;
        
         for(int i=0;i<nums.size();i++)
         {
             if(i%2==0)
                 even+=nums[i];
             else
                 odd+=nums[i];
         }
         int even1=0,odd1=0,ans=0;
         for(int i=0;i<nums.size();i++)
         {
             if(i%2==0)
             {
                 if(even1+odd-odd1 == odd1+even-even1-nums[i])
                     ans++;
                 
                 even1+=nums[i];
                     
             }
             else
             {
                 if(even1+odd-odd1-nums[i] == odd1+even-even1)
                     ans++;
                 
                 odd1+=nums[i];
             }
         }
        return ans;
    }
};

File: Array/ways-to-make-a-fair-array.py
class Solution:
    def waysToMakeFair(self, nums: List[int]) -> int:
        s1, s2 = [0, 0], [sum(nums[0::2]), sum(nums[1::2])]
        res = 0
        for i, a in enumerate(nums):
            s2[i % 2] -= a
            res += s1[0] + s2[1] == s1[1] + s2[0]
            s1[i % 2] += a
        return res
            